relayer/relayer.go

```diff
@@ -158,63 +164,107 @@ func NewRelayer(
 	return &r, sub, nil
 }
 
-func (r *Relayer) processMissedBlocks(
+// Determines the height to process from. There are two cases:
+// 1) The database contains the latest processed block data for the chain
+//   - In this case, we return the maximum of the latest processed block and the configured start block height
+//
+// 2) The database has been configured for the chain, but does not contain the latest processed block data
+//   - In this case, we return the configured start block height
+func (r *Relayer) calculateStartingBlockHeight(
 	sub vms.Subscriber,
-) error {
-	// Get the latest processed block height from the database.
-	latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
-
-	// The following cases are treated as successful:
-	// 1) The database contains the latest processed block data for the chain
-	//    - In this case, we parse the block height and process warp logs from that height to the current block
-	// 2) The database has been configured for the chain, but does not contain the latest processed block data
-	//    - In this case, we save the current block height in the database, but do not process any historical warp logs
-	if err == nil {
-		// If the database contains the latest processed block data, then back-process all warp messages from that block to the latest block
-		// Note that the retrieved latest processed block may have already been partially (or fully) processed by the relayer on a previous run. When
-		// processing a warp message in real time, which is when we update the latest processed block in the database, we have no way of knowing
-		// if that is the last warp message in the block
-		latestProcessedBlock, success := new(big.Int).SetString(string(latestProcessedBlockData), 10)
-		if !success {
-			r.logger.Error("failed to convert latest block to big.Int", zap.Error(err))
-			return err
-		}
+	startBlockHeight uint64,
+) (*big.Int, error) {
+	var height *big.Int
```

Comment:
Do we still need to use `big.Int` here, or would it be possible to simplify it to use `uint64`?

Reply:
Either way, I'd recommend removing the `height` result variable in favor returning the proper value directly in each case down below. Looks like it's only assigned immediately before returning it.

Reply:
Simplified to use uint64